### PR TITLE
dev: migrate to use ethereum-types

### LINKED ISSUE #230: dev: update to use `ethereum-types`
EELS is no longer using a local `ethereum.base_types` module, but the `ethereum_types` package.

### PR DESCRIPTION
**Pull request by enitrat**
Closes #230

@ClementWalter: do you want to migrate the cairo files to a `ethereum_types.{numeric | bytes}` module?
if so - what about the compound types - do we create a third file ?

### CODE REVIEW

**Comment by ClementWalter** [comment]
seems like some types have changed looking at the failed CI

File: cairo/pyproject.toml [hunk 1]
```diff
@@ -8,6 +8,7 @@ requires-python = ">=3.10"
 dependencies = [
   "cairo-lang>=0.13.2",
   "ethereum",
+  "ethereum-types>=0.2.3,<0.3",
```
[the remark this answers is not in the document]
Comment (enitrat) [reply cairo/pyproject.toml:11]: meaning we shouldn't constrain this ourselves ?

File: cairo/pyproject.toml [hunk 2]
```diff
@@ -8,6 +8,7 @@ requires-python = ">=3.10"
 dependencies = [
   "cairo-lang>=0.13.2",
   "ethereum",
+  "ethereum-types>=0.2.3",
```
Comment (ClementWalter) [inline cairo/pyproject.toml:11]: do we need this?

**Comment by enitrat** [comment]
CI is ok, just this weird error... would be nice to merge soon so as to not be forced to use the "old" types in next PRs.

**Review by obatirou — APPROVED** [review]

**Pull request merged**